File: src/types/workbench.ts
export type WorkbenchSectionStatus = "available" | "empty" | "forbidden" | "error";

export interface WorkbenchTodoItemDTO {
  key: string;
  count: number;
  severity: string;
  route_key: string;
  action_key: string;
}

export interface WorkbenchTodosSectionDTO {
  status: WorkbenchSectionStatus;
  error_code: string | null;
  items: WorkbenchTodoItemDTO[];
  total: number;
}

export type WorkbenchTaskStatus =
  | "needs_action"
  | "submitted"
  | "processing"
  | "completed"
  | "partial"
  | "failed";
export type WorkbenchTaskPriority = "urgent" | "high" | "normal" | "low";

export interface WorkbenchTaskItemDTO {
  task_ref: string;
  task_type: string;
  object_name: string;
  project_name: string | null;
  status: WorkbenchTaskStatus;
  priority: WorkbenchTaskPriority;
  assignee: string;
  responsibility: string;
  created_at: string | null;
  updated_at: string | null;
  waiting_minutes: number | null;
  next_action_key: string | null;
  next_action_label: string;
  route_key: string | null;
  result_summary: string | null;
  progress_total: number | null;
  progress_success: number | null;
  progress_failed: number | null;
}

export interface WorkbenchTaskCenterSectionDTO {
  status: WorkbenchSectionStatus;
  error_code: string | null;
  summary: {
    needs_action: number;
    running: number;
    attention: number;
    completed_today: number;
  };
  priority_items: WorkbenchTaskItemDTO[];
  my_tasks: WorkbenchTaskItemDTO[];
  running_jobs: WorkbenchTaskItemDTO[];
  attention_items: WorkbenchTaskItemDTO[];
  recent_completed: WorkbenchTaskItemDTO[];
}

export interface WorkbenchOperationCardDTO {
  key: string;
  label: string;
  count: number;
  severity: string;
  action_hint: string | null;
  scope?: string | null;
  project_id?: string | null;
  context_label?: string | null;
}

export interface WorkbenchOperationsDataDTO {
  title_visible: boolean;
  scope: string;
  window_days: number;
  cards: WorkbenchOperationCardDTO[];
  indexing: {
    index_failed: number;
    skipped: number;
    not_indexed: number;
    parse_failed: number;
    parse_pending: number;
    parse_processing: number;
    kb_init_failed: number;
  };
  access: {
    pending_original_requests: number;
    overdue_original_requests: number;
    recent_auto_approved: number;
    timeout_enabled: boolean;
  };
  lifecycle: {
    archive_candidates: number;
    archive_warnings: number;
    needs_update: number;
    reuse_upgrade_candidates: number;
  };
}

export interface WorkbenchOperationsSectionDTO {
  status: WorkbenchSectionStatus;
  error_code: string | null;
  data: WorkbenchOperationsDataDTO | null;
}

export interface WorkbenchProjectItemDTO {
  project_id: string;
  name: string;
  status: string;
  project_role: string;
  lifecycle_route_key: string | null;
  lifecycle_phase_key: string | null;
}

export interface WorkbenchProjectsSectionDTO {
  status: WorkbenchSectionStatus;
  error_code: string | null;
  items: WorkbenchProjectItemDTO[];
  total: number;
}

export interface WorkbenchRecentActivityItemDTO {
  asset_id: string;
  title: string;
  scope: string;
  zone: string;
  asset_type: string;
  confidentiality_level: string;
  summary: string | null;
  project_name: string | null;
  updated_at: string | null;
}

export interface WorkbenchRecentActivitySectionDTO {
  status: WorkbenchSectionStatus;
  error_code: string | null;
  items: WorkbenchRecentActivityItemDTO[];
  total: number;
}

export interface WorkbenchOverviewDTO {
  task_center: WorkbenchTaskCenterSectionDTO;
  todos: WorkbenchTodosSectionDTO;
  operations: WorkbenchOperationsSectionDTO;
  projects: WorkbenchProjectsSectionDTO;
  recent_activity: WorkbenchRecentActivitySectionDTO;
}

File: backend/app/schemas/workbench.py
"""First-party browser workbench response contract.

The contract is intentionally separate from Agent Gateway schemas and exposes
only first-party business identifiers and permission-filtered projections.
"""

from __future__ import annotations

import uuid
from datetime import datetime
from enum import Enum

from pydantic import BaseModel, Field

from app.schemas.knowledge_insights import (
    AccessInsights,
    InsightCard,
    LifecycleInsights,
)


class WorkbenchSectionStatus(str, Enum):
    available = "available"
    empty = "empty"
    forbidden = "forbidden"
    error = "error"


class WorkbenchTodoItem(BaseModel):
    key: str
    count: int
    severity: str
    route_key: str
    action_key: str


class WorkbenchTodosSection(BaseModel):
    status: WorkbenchSectionStatus
    error_code: str | None = None
    items: list[WorkbenchTodoItem] = Field(default_factory=list)
    total: int = 0


class WorkbenchTaskItem(BaseModel):
    """Safe, permission-filtered task projection shared by home and the global drawer."""

    task_ref: str
    task_type: str
    object_name: str
    project_name: str | None = None
    status: str
    priority: str
    assignee: str
    responsibility: str
    created_at: datetime | None = None
    updated_at: datetime | None = None
    waiting_minutes: int | None = None
    next_action_key: str | None = None
    next_action_label: str
    route_key: str | None = None
    result_summary: str | None = None
    progress_total: int | None = None
    progress_success: int | None = None
    progress_failed: int | None = None


class WorkbenchTaskSummary(BaseModel):
    needs_action: int = 0
    running: int = 0
    attention: int = 0
    completed_today: int = 0


class WorkbenchTaskCenterSection(BaseModel):
    status: WorkbenchSectionStatus
    error_code: str | None = None
    summary: WorkbenchTaskSummary = Field(default_factory=WorkbenchTaskSummary)
    priority_items: list[WorkbenchTaskItem] = Field(default_factory=list)
    my_tasks: list[WorkbenchTaskItem] = Field(default_factory=list)
    running_jobs: list[WorkbenchTaskItem] = Field(default_factory=list)
    attention_items: list[WorkbenchTaskItem] = Field(default_factory=list)
    recent_completed: list[WorkbenchTaskItem] = Field(default_factory=list)


class WorkbenchOperationsIndexing(BaseModel):
    index_failed: int = 0
    skipped: int = 0
    not_indexed: int = 0
    parse_failed: int = 0
    parse_pending: int = 0
    parse_processing: int = 0
    kb_init_failed: int = 0


class WorkbenchOperationsSummary(BaseModel):
    title_visible: bool
    scope: str
    window_days: int
    cards: list[InsightCard] = Field(default_factory=list)
    indexing: WorkbenchOperationsIndexing
    access: AccessInsights
    lifecycle: LifecycleInsights


class WorkbenchOperationsSection(BaseModel):
    status: WorkbenchSectionStatus
    error_code: str | None = None
    data: WorkbenchOperationsSummary | None = None


class WorkbenchProjectItem(BaseModel):
    project_id: uuid.UUID
    name: str
    status: str
    project_role: str
    lifecycle_route_key: str | None = None
    lifecycle_phase_key: str | None = None


class WorkbenchProjectsSection(BaseModel):
    status: WorkbenchSectionStatus
    error_code: str | None = None
    items: list[WorkbenchProjectItem] = Field(default_factory=list)
    total: int = 0


class WorkbenchRecentActivityItem(BaseModel):
    asset_id: uuid.UUID
    title: str
    scope: str
    zone: str
    asset_type: str
    confidentiality_level: str
    summary: str | None = None
    project_name: str | None = None
    updated_at: datetime | None = None


class WorkbenchRecentActivitySection(BaseModel):
    status: WorkbenchSectionStatus
    error_code: str | None = None
    items: list[WorkbenchRecentActivityItem] = Field(default_factory=list)
    total: int = 0


class WorkbenchOverviewResponse(BaseModel):
    task_center: WorkbenchTaskCenterSection
    todos: WorkbenchTodosSection
    operations: WorkbenchOperationsSection
    projects: WorkbenchProjectsSection
    recent_activity: WorkbenchRecentActivitySection
